db/migrate/20240819045115_migrate_unit_size_to_variants.rb

```diff
@@ -0,0 +1,21 @@
+class MigrateUnitSizeToVariants < ActiveRecord::Migration[7.0]
+   def up
+    # Copy variant_unit only if it's empty in the variant
+    ActiveRecord::Base.connection.execute(<<-SQL
+      UPDATE spree_variants
+      SET variant_unit = spree_products.variant_unit
+      FROM spree_products
+      WHERE spree_variants.product_id = spree_products.id
+      AND spree_variants.variant_unit IS NULL
+    SQL
+    )
+
+    ActiveRecord::Base.connection.execute(<<-SQL
+      UPDATE spree_variants
+      SET variant_unit_scale = spree_products.variant_unit_scale, variant_unit_name = spree_products.variant_unit_name
+      FROM spree_products
+      WHERE spree_variants.product_id = spree_products.id
+    SQL
+    )
```

Comment:
`variant_unit` already existed on both `product` and `variant` table, the condition is here to not override the value in the variant table. I don't think it's necessary for the other new fileds.

Reply:
Oh.. I must have missed that.. 
So if you can only select `variant_unit` on the product, and the `variant` is the source of truth, the code is already assuming these are in sync. So in theory, no variant_unit values will not need to be migrated (but it's still a good idea to have the migration just in case).